node/core/prospective-parachains/src/lib.rs

```diff
@@ -429,22 +432,35 @@ fn answer_get_backable_candidate(
 fn answer_hypothetical_depths_request(
 	view: &View,
 	request: HypotheticalDepthRequest,
-	tx: oneshot::Sender<Vec<usize>>,
+	tx: oneshot::Sender<Vec<(usize, ProspectiveCandidateParentState)>>,
 ) {
-	match view
+	let fragment_tree = view
 		.active_leaves
 		.get(&request.fragment_tree_relay_parent)
-		.and_then(|l| l.fragment_trees.get(&request.candidate_para))
-	{
-		Some(fragment_tree) => {
-			let depths = fragment_tree.hypothetical_depths(
-				request.candidate_hash,
-				request.parent_head_data_hash,
-				request.candidate_relay_parent,
-			);
+		.and_then(|l| l.fragment_trees.get(&request.candidate_para));
+	let candidate_storage = view.candidate_storage.get(&request.candidate_para);
+	match (fragment_tree, candidate_storage) {
+		(Some(fragment_tree), Some(candidate_storage)) => {
+			let depths = fragment_tree
+				.hypothetical_depths(
+					request.candidate_hash,
+					request.parent_head_data_hash,
+					request.candidate_relay_parent,
+				)
+				.into_iter()
+				.map(|(depth, parent)| {
+					let parent_status = if depth == 0 || candidate_storage.is_backed(&parent) {
```

Comment:
`is_backed` would return `false` for candidate not present in storage, but we do expect fragment trees and candidate storage to be updated correspondingly, thus this should never happen